pkg/pgsql/server/fmessages/bind.go

```diff
@@ -0,0 +1,150 @@
+// name of the source prepared statement (empty string denotes the unnamed prepared statement), the name of the destination
+// portal (empty string denotes the unnamed portal), and the values to use for any parameter placeholders present in the
+// prepared statement. The supplied parameter set must match those needed by the prepared statement. (If you declared
+// any void parameters in the Parse message, pass NULL values for them in the Bind message.) Bind also specifies
+// the format to use for any data returned by the query; the format can be specified overall, or per-column.
+// The response is either BindComplete or ErrorResponse.
+//
+// Note
+// The choice between text and binary output is determined by the format codes given in Bind, regardless of the SQL
+// command involved. The BINARY attribute in cursor declarations is irrelevant when using extended query protocol.
+type BindMsg struct {
+	// The name of the destination portal (an empty string selects the unnamed portal).
+	DestPortalName string
+	// The name of the source prepared statement (an empty string selects the unnamed prepared statement).
+	PreparedStatementName string
+	// The number of parameter format codes that follow (denoted C below). This can be zero to indicate that there are no parameters or that the parameters all use the default format (text); or one, in which case the specified format code is applied to all parameters; or it can equal the actual number of parameters.
+	//ParametersCountNumber int16
+	// The parameter format codes. Each must presently be zero (text) or one (binary).
+	ParameterFormatCodes []int16
+	// The number of parameter values that follow (possibly zero). This must match the number of parameters needed by the query.
+	//ParametersValueCount int16
+	// Array of the values of the parameters, in the format indicated by the associated format code. n is the above length.
+	ParamVals []interface{}
+	// The number of result-column format codes that follow (denoted R below). This can be zero to indicate that there are no result columns or that the result columns should all use the default format (text); or one, in which case the specified format code is applied to all result columns (if any); or it can equal the actual number of result columns of the query.
+	//ResultColumnFormatCodesNumber int16
+	// The result-column format codes. Each must presently be zero (text) or one (binary).
+	ResultColumnFormatCodes []int16
+}
+
+func ParseBindMsg(payload []byte) (BindMsg, error) {
+	b := bytes.NewBuffer(payload)
+	r := bufio.NewReader(b)
+	destPortalName, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	preparedStatement, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+
+	parameterFormatCodeNumber, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	parameterFormatCodes := make(map[int]int16, 0)
```

Comment:
Is a map needed here? We're storing those ints sequentially  so a slice should be enough